apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/json/JsonReaders.kt

```diff
@@ -0,0 +1,66 @@
+@file:JvmName("-JsonReaders")
+package com.apollographql.apollo3.api.json
+
+import com.apollographql.apollo3.annotations.ApolloInternal
+import okio.BufferedSource
+import kotlin.jvm.JvmName
+
+fun BufferedSource.jsonReader(): JsonReader {
+  return BufferedSourceJsonReader(this)
+}
+
+fun Map<String, Any?>.jsonReader(): JsonReader {
+  return MapJsonReader(this)
+}
+
+/**
+ * Reads the reader and maps numbers to the closest representation possible in that order:
+ * - Int
+ * - Long
+ * - Double
+ * - JsonNumber
+ */
+@ApolloInternal
+fun JsonReader.readAny(): Any? {
```

Comment:
This could very well become public API at some point but wasn't just sure yet so kept the `@ApolloInternal` annotation.